compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2183,6 +2191,31 @@ object Parsers {
         commaSeparated(() => argType())
     end argTypes
 
+    def argCaps(namedOK: Boolean, wildOK: Boolean, tupleOK: Boolean): List[Tree] = // TODO grammar doc
+      def argType() =
+        val t = concreteCapsType(captureSetOrRef())
+        if wildOK then t else rejectWildcardType(t) // TODO needed?
+
+      def namedArgType() =
+        atSpan(in.offset):
+          val name = ident()
+          accept(EQUALS)
+          NamedArg(name.toTypeName, argType())
+
+      def namedElem() =
+        atSpan(in.offset):
+          val name = ident()
+          acceptColon()
+          NamedArg(name, argType())
+
+      if namedOK && isIdent && in.lookahead.token == EQUALS then
+        commaSeparated(() => namedArgType())
+      else if tupleOK && isIdent && in.lookahead.isColon && in.featureEnabled(Feature.namedTuples) then
```

Comment:
@bracevac Here is where it goes wrong. The `Feature.namedTuples` method doesn't exist anymore because named tuples were stabilized in `main` (#22753). Consider the following change:
```suggestion
      else if tupleOK && isIdent && in.lookahead.isColon && sourceVersion.enablesNamedTuples then
```